ote_cli/ote_cli/tools/explain.py

```diff
@@ -0,0 +1,160 @@
+"""
+Model inference demonstration tool.
+"""
+
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import argparse
+import os
+import time
+from collections import deque
+
+import cv2
+import numpy as np
+from ote_sdk.configuration.helper import create
+from ote_sdk.entities.subset import Subset
+from ote_sdk.entities.inference_parameters import InferenceParameters
+from ote_sdk.entities.task_environment import TaskEnvironment
+
+from ote_cli.datasets import get_dataset_class
+from ote_cli.registry import find_and_parse_model_template
+from ote_cli.utils.config import override_parameters
+from ote_cli.utils.importing import get_impl_class
+from ote_cli.utils.io import get_image_files, read_label_schema, read_model, save_saliency_output
+from ote_cli.utils.parser import (
+    add_hyper_parameters_sub_parser,
+    gen_params_dict_from_args,
+)
+
+ESC_BUTTON = 27
+SUPPORTED_EXPLAIN_ALGORITHMS = ["CAM", "EigenCAM"]
+
+
+def parse_args():
+    """
+    Parses command line arguments.
+    """
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument("template")
+    parsed, _ = pre_parser.parse_known_args()
+    # Load template.yaml file.
+    template = find_and_parse_model_template(parsed.template)
+    # Get hyper parameters schema.
+    hyper_parameters = template.hyper_parameters.data
+    assert hyper_parameters
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("template")
+    parser.add_argument(
+        "-i",
+        "--input",
```

Comment:
to be consistent with `ote train` and `ote eval` propose `--input` -> `--explain-data`